### PR TITLE
Fix crash when roomSummary is nil

### PR DESCRIPTION
**Pull request by Velin92**
This PR fixes the following crash: https://github.com/vector-im/element-ios/issues/7403

### CODE REVIEW

File: MatrixSDK/Data/RoomList/Common/MXBreadcrumbsRoomListDataFetcher.swift [hunk 1]
```diff
@@ -91,7 +91,7 @@ internal class MXBreadcrumbsRoomListDataFetcher: NSObject, MXRoomListDataFetcher
         
         if let query = fetchOptions.filterOptions.query?.lowercased(), !query.isEmpty {
             recentRoomIds = recentRoomIds.filter({ roomId in
-                guard let displayname = session?.roomSummary(withRoomId: roomId).displayname else {
+                guard let displayname = session?.roomSummary(withRoomId: roomId)?.displayname else {
```
Comment (stefanceriu) [inline MatrixSDK/Data/RoomList/Common/MXBreadcrumbsRoomListDataFetcher.swift:94]: Pet peve, but can we rename this to `displayName` please?

File: MatrixSDK/MXSession.h [hunk 2]
```diff
@@ -1087,7 +1087,7 @@ typedef void (^MXOnBackgroundSyncFail)(NSError *error);
 
  @return the MXRoomSummary instance.
  */
-- (MXRoomSummary *)roomSummaryWithRoomId:(NSString*)roomId;
+- (nullable MXRoomSummary *)roomSummaryWithRoomId:(NSString*)roomId;
```
Comment (stefanceriu) [inline MatrixSDK/MXSession.h:1090]: I'm pretty sure this nullability specifier will trigger a ton of the warnings for the rest of the methods in this file. Have a look at `MX_ASSUME_MISSING_NULLABILITY_BEGIN`

**Review by stefanceriu — APPROVED** [review]
Looks good to me except for the inline comments

File: MatrixSDK/MXSession.h (hunk 2 above)
Comment (Velin92) [reply MatrixSDK/MXSession.h:1090]: Oh we don't have it in MXSession.h

**Pull request merged**